### PR TITLE
Added clickOn longClickOn wihtId and withText

### PR DESCRIPTION
**Pull request by jamesbluecrow**

### CODE REVIEW

**Comment by jamesbluecrow** [comment]
This PR tries to fix an issue https://github.com/SchibstedSpain/Barista/issues/195

File: library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt [hunk 1]
```diff
@@ -26,72 +28,84 @@ import org.hamcrest.Matchers.anyOf
...
+  fun clickOn(@IdRes id: Int, text: String) {
+    performClickTypeOnMatcher(allOf(withId(id), withText(text)), click())
+  }
 
-    @JvmStatic
-    fun longClickOn(text: String) {
-        performClickTypeOnMatcher(withText(text), longClick())
-    }
+  @JvmStatic
+  fun longClickOn(resId: Int) {
```
Comment (SmasSive) [inline library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt]: Should be `@IdRes id: Int` just to normalize the parameters name

File: library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt [hunk 2]
```diff
@@ -26,72 +28,84 @@ import org.hamcrest.Matchers.anyOf
...
+  fun longClickOn(text: String) {
+    performClickTypeOnMatcher(withText(text), longClick())
+  }
+
+  @JvmStatic
+  fun longClickOn(@IdRes id: Int, text: String) {
+    performClickTypeOnMatcher(allOf(withId(id), withText(text)), longClick())
+  }
+
+  private fun performClickTypeOnMatcher(viewMatcher: Matcher<View>, clickType: ViewAction) {
```
Comment (SmasSive) [inline library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt]: This code should be refactored once #194 (or @Sloy proposal) is merged

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/LonfClickIdAndTextTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed;
+import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertNotExist;
+import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.longClickOn;
+
+@RunWith(AndroidJUnit4.class)
+public class LonfClickIdAndTextTest {
```
Comment (SmasSive) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/LonfClickIdAndTextTest.java]: Typo here `LonfClickIdAndTextTest` should be `LongClickIdAndTextTest`

File: library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt (hunk 1 above)
Comment (Sloy) [reply library/src/main/java/com/schibsted/spain/barista/interaction/BaristaClickInteractions.kt]: Goto https://github.com/SchibstedSpain/Barista/issues/158 😇

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/LonfClickIdAndTextTest.java (hunk 3 above)
Comment (jamesbluecrow) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/LonfClickIdAndTextTest.java]: Nice catch!

**Comment by rocboronat** [comment]
I missed your birthday, PR 😞

**Comment by Sloy** [comment]
Oh :( 
Sorry about that @jamesbluecrow, we should have pushed this more.

I'm going to close the PR, since https://github.com/SchibstedSpain/Barista/issues/224 should be a more flexible approach that also solves your mentioned issue (#195).

**Pull request closed without merging**